binding/request_binder.go

```diff
@@ -0,0 +1,111 @@
+package binding
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gobuffalo/buffalo/internal/httpx"
+	"github.com/gobuffalo/nulls"
+	"github.com/monoculum/formam"
+)
+
+var (
+	errBlankContentType = errors.New("blank content type")
+)
+
+// RequestBinder is in charge of binding multiple requests types to
+// structs.
+type RequestBinder struct {
+	lock    *sync.Mutex
+	binders map[string]Binder
```

Comment:
consider using a `sync.Map`? You can use https://github.com/gobuffalo/mapgen to generate a typed version of the map. its a little cleaner to use than  the mutex.

Reply:
Did not get this one. Will ask you offline.